clang/test/SemaSYCL/invalid-kernel-arguments.cpp

```diff
@@ -48,6 +48,9 @@ class H {
   int A;
 };
 
+int GlobalNoAttribute;
```

Comment:
You're missing tests that it applies only to global variables (not other things like functions, or local variables) and tests that it accepts no arguments.

Reply:
Latest commit added new test file for checking this attribute specifically. I included the additional checks you mentioned.